cdc/processor.go

```diff
@@ -1252,6 +1273,65 @@ func (p *oldProcessor) sorterConsume(
 	}
 }
 
+// shouldSplitUpdateEventRow determines if the split event is needed to align the old format based on
+// whether the handle key column has been modified.
+// If the handle key column is modified,
+// we need to use splitUpdateEventRow to split the row update event into a row delete and a row insert event.
+func shouldSplitUpdateEventRow(rowUpdateChangeEvent *model.RowChangedEvent) bool {
+	// nil row will never be split.
+	if rowUpdateChangeEvent == nil {
+		return false
+	}
+
+	handleKeyCount := 0
+	equivalentHandleKeyCount := 0
+	for i := range rowUpdateChangeEvent.Columns {
+		if rowUpdateChangeEvent.Columns[i].Flag.IsHandleKey() && rowUpdateChangeEvent.PreColumns[i].Flag.IsHandleKey() {
+			handleKeyCount++
+			colValueString := model.ColumnValueString(rowUpdateChangeEvent.Columns[i].Value)
+			preColValueString := model.ColumnValueString(rowUpdateChangeEvent.PreColumns[i].Value)
+			if colValueString == preColValueString {
+				equivalentHandleKeyCount++
+			}
+		}
+	}
+
+	// If the handle key columns are not updated, so we do **not** need to split the event row.
+	return !(handleKeyCount == equivalentHandleKeyCount)
+}
+
+// splitUpdateEventRow splits a row update event into a row delete and a row insert event.
+func splitUpdateEventRow(rowUpdateChangeEvent *model.RowChangedEvent) ([]*model.RowChangedEvent, error) {
```

Comment:
Wait for addressing https://github.com/pingcap/ticdc/pull/2367#discussion_r677938414

Reply:
Addressed. Thanks for your review!